node/utils/checkAndUpdateSchemas.ts

```diff
@@ -0,0 +1,132 @@
+} from '../../common/constants'
+
+const appVersion = process.env.VTEX_APP_VERSION as string
+const BUCKET_NAME = 'rma-schemaUpdated'
+
+const isSchemaUpdated = async (ctx: Context) => {
+  if (LINKED) return false
+  const {
+    clients: { vbase },
+  } = ctx
+
+  const wasUpdated = await vbase.getJSON<boolean | null>(
+    BUCKET_NAME,
+    appVersion,
+    true
+  )
+
+  if (wasUpdated) return true
+
+  return false
+}
+
+export const checkAndUpdateSchemas = async (ctx: Context) => {
+  const {
+    clients: { vbase, mdFactory },
+    vtex: { logger },
+  } = ctx
+
+  let hasUpdated = false
+
+  try {
+    const updated = await isSchemaUpdated(ctx)
+
+    // if updated === true, we don't need to update the schema
+    if (updated) return
+
+    // we update vbase no matter the result of hasChanged, so next call we skip the check
+    await vbase.saveJSON(BUCKET_NAME, appVersion, 'true')
+
+    // if hasChanged === false, we don't need to update the schema
+    const settingsSchemaPromise = mdFactory.getSettingsSchema()
+    const requestSchemaPromise = mdFactory.getRequestSchema()
+    const commentsSchemaPromise = mdFactory.getCommentsSchema()
+    const historySchemaPomise = mdFactory.getHistorySchema()
+    const productsSchemaPromise = mdFactory.getProductsSchema()
+
+    const [
+      settingsSchema,
+      requestSchema,
+      commentsSchema,
+      historySchema,
+      productsSchema,
+    ] = await Promise.all([
+      settingsSchemaPromise,
+      requestSchemaPromise,
+      commentsSchemaPromise,
+      historySchemaPomise,
+      productsSchemaPromise,
+    ])
+
+    let updateSettingsSchemaPromise
+    let updateRequestSchemaPromise
+    let updateCommentsSchemaPromise
+    let updateHistorySchemaPromise
+    let updateProductsSchemaPromise
+
+    if (!isDeepStrictEqual(settingsSchema, SETTINGS_SCHEMA)) {
+      updateSettingsSchemaPromise =
+        mdFactory.updateSettingsSchema(SETTINGS_SCHEMA)
+      hasUpdated = true
+    }
+
+    if (!isDeepStrictEqual(requestSchema, RETURNS_SCHEMA)) {
+      updateRequestSchemaPromise = mdFactory.updateRequestSchema(RETURNS_SCHEMA)
+      hasUpdated = true
+    }
+
+    if (!isDeepStrictEqual(commentsSchema, COMMENTS_SCHEMA)) {
+      updateCommentsSchemaPromise =
+        mdFactory.updateCommentsSchema(COMMENTS_SCHEMA)
+      hasUpdated = true
+    }
+
+    if (!isDeepStrictEqual(historySchema, HISTORY_SCHEMA)) {
+      updateHistorySchemaPromise = mdFactory.updateHistorySchema(HISTORY_SCHEMA)
+      hasUpdated = true
+    }
+
+    if (!isDeepStrictEqual(productsSchema, PRODUCTS_SCHEMA)) {
+      updateProductsSchemaPromise =
+        mdFactory.updatetProductsSchema(PRODUCTS_SCHEMA)
+      hasUpdated = true
+    }
+
+    await Promise.all([
+      updateSettingsSchemaPromise,
+      updateRequestSchemaPromise,
+      updateCommentsSchemaPromise,
+      updateHistorySchemaPromise,
+      updateProductsSchemaPromise,
+    ])
+  } catch (error) {
+    if (error.response.status !== 304) {
```

Comment:
@filafb @zekivtex @cesarocampov 
Would be there any advantage trying to catch errors from independent promises, to later log or return?

I was thinking like, logging which schema failed and why, and have a Splunk trigger if this is happening in a `master` workspace, as a red flag for us to contact the owners of missuse 🚩

Reply:
Correct me if I am wrong, but to catch independent errors, we need to run the calls synchronously. Doing so, the tradeoff can be very expensive. On postman, a call to update a schema can take a couple of seconds. If we make the calls synchronous, that time will pilled up. An alternative, could be to use `Promise.allSettle`, but TS doesn't allow me to use that (and tbh, I am not full aware about its performance).

And in theory, the errors regarding a specific schema should happen during development, since the only missuse I see here is on a dev side.

Reply:
With promise.all the first error would trigger the catch, and stop the rest; I was thinking of logging which one was the problematic one, so If you are developing modifying more than 1 schema, you could see the exact one. But maybe there's no need of knowing that haha

Reply:
But is there a way to do that without running the requests synchronously? I think it would be a nice addition, if it doesn't hurt the performance.

Reply:
Yeah! The methods from the mdFactory are not catching errors, if we catch and throw them there, each method can have its own custom error object passed along to the promise.all catcher. There we could see who's error it's when using logger.

Or we could also write a catch, for every one of them independently, when assigning them to the `updateSchemaPromise`. That error object would be passed along also. But that will mean that .all will behave exactly as the sugar coated allSettled and not halt with the first error